code/Model/Submission/SubmittedFileField.php

```diff
@@ -41,14 +41,24 @@ public function getFormattedValue()
         $name = $this->getFileName();
         $link = $this->getLink();
         $title = _t(__CLASS__ . '.DOWNLOADFILE', 'Download File');
+        $message = _t(__CLASS__ . '.INSUFFICIENTRIGHTSTOVIEW', 'Insufficient rights to view the file');
+        $file = $this->getUploadedFileFromDraft();
 
         if ($link) {
-            return DBField::create_field('HTMLText', sprintf(
-                '%s - <a href="%s" target="_blank">%s</a>',
-                $name,
-                $link,
-                $title
-            ));
+            if ($file->canView()) {
+                return DBField::create_field('HTMLText', sprintf(
+                    '%s - <a href="%s" target="_blank">%s</a>',
+                    $name,
+                    $link,
+                    $title
```

Comment:
There isn't a risk of XSS here, but by explicitly escaping everything we make this explicit. 

```suggestion
                    htmlspecialchars($name),
                    htmlspecialchars($link),
                    htmlspecialchars($title)
```

Reply:
DONE